osv2/mget_test.go

```diff
@@ -0,0 +1,312 @@
+			name: "Empty Request",
+			fields: fields{
+				Index: "",
+				Docs:  []opensearchtools.RoutableDoc{},
+			},
+			want:    []byte(`{"docs":[]}`),
+			wantErr: false,
+		},
+		{
+			name: "Single document single type",
+			fields: fields{
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{id: testID1, index: testIndex1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Multiple documents single type",
+			fields: fields{
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{id: testID1, index: testIndex1},
+					mgetTestDoc{id: testID2, index: testIndex2},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"},{"_id":"test_id2","_index":"test_index2"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Multiple documents mixed type",
+			fields: fields{
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{id: testID1, index: testIndex1},
+					opensearchtools.NewDocumentRef(testIndex2, testID2),
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"},{"_id":"test_id2","_index":"test_index2"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Document without index",
+			fields: fields{
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{id: testID1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Document without id",
+			fields: fields{
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{index: testIndex1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"","_index":"test_index"}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Document without id and index",
+			fields: fields{
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":""}]}`),
+			wantErr: false,
+		},
+		{
+			name: "Request level index does not affect request json body",
+			fields: fields{
+				Index: testIndex2,
+				Docs: []opensearchtools.RoutableDoc{
+					mgetTestDoc{id: testID1, index: testIndex1},
+				},
+			},
+			want:    []byte(`{"docs":[{"_id":"test_id","_index":"test_index"}]}`),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := &opensearchtools.MGetRequest{
+				Index: tt.fields.Index,
+				Docs:  tt.fields.Docs,
+			}
+			marshalableMGetRequest := FromModelMGetRequest(m)
+			got, err := marshalableMGetRequest.MarshalJSON()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MarshalJSON() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			require.JSONEq(t, string(tt.want), string(got))
+		})
+	}
+}
+
+func Test_MGetResult_ToModel(t *testing.T) {
+	type fields struct {
+		Index       string
+		ID          string
+		Version     int
+		SeqNo       int
+		PrimaryTerm int
+		Found       bool
+		Source      json.RawMessage
+		Error       error
+	}
+	tests := []struct {
+		name   string
+		fields fields
```

Comment:
Do you like the auto generated `fields` pattern? I personally don't and would just have a `target MGetResult` attribute instead.

Reply:
Ya, I agree. It seems pointless.

Reply:
addressed this in another commit